partition/roles/sonic-config/README.md

```diff
@@ -0,0 +1,513 @@
+> If you are using the sonic role check the [Migration](#migration-from-sonic-role) section below.
+
+This role creates a `config_db.json` on a SONiC switch, configures the timezone and adds a `resolv.conf`.
+Optionally it may render a `frr.conf` for routing or an `iptables.json` for extended CACL.
+
+## Supported SONiC Versions
+
+The only supported SONiC versions are Edgecore SONiC versions 202111.x.
+Other versions might work as well but their behavior might slightly differ for some of the configuration parameters.
+For example, if you are using `sonic_config_docker_routing_config_mode: split` on a 202111.x version and wish to migrate to version 202311.5, you will need to use `split-unified` instead.
+As the 202111.x versions worked best for us so far this role is mostly tested against those versions.
+
+## Problems with DHCP relay
+
+Some of our deployments require a DHCP relay agent on the switch which relays PXE boot requests between a client and a PXE server.
+From version 202111.6 to version 202111.7 some change in the dhcp_relay broke the relay's ability to forward PXE boot OFFER messages from the server back to the client.
+This bug was fixed in version 202111.11.
+
+## Migration from the Deprecated `sonic` Role
+
+Most of the variables have the same name, meaning and structure as the corresponding ones in the sonic role.
+So in most cases simply adding `_config` to the variable name is enough, e.g. `sonic_interconnects` becomes `sonic_config_interconnects`.
+Variables that were changed, removed or added are explained below.
+
+### `sonic_ntpservers`
+
+Change
+
+```yaml
+sonic_ntpservers:
+  - 1.2.3.4
+  - 5.6.7.8
+```
+
+to
+
+```yaml
+sonic_config_ntp:
+  servers:
+    - 1.2.3.4
+    - 5.6.7.8
+
+  # Additionally source interface and VRF can be configured.
+  src_interface: # Loopback0
+  vrf: # default
+```
+
+### sonic_mgmtif_ip and sonic_mgmtif_gateway
+
+Change
+
+```yaml
+sonic_mgmtif_ip: 10.1.2.3
+sonic_mgmtif_gateway: 10.1.2.2
+```
+
+to
+
+```yaml
+sonic_config_mgmt_interface:
+  gateway_address: 10.1.2.2
+  ip: 10.1.2.3
+```
+
+### sonic_ip_masquerade
+
+This variable was removed.
+See [this mini-lab PR](https://github.com/metal-stack/mini-lab/pull/237).
+
+### sonic_config_action
+
+This variable is replaced by the boolean variable `sonic_config_reload_config` which defaults to `false`.
+See [below](#variables) for explanation.
+
+### sonic_ports, sonic_ports_default_speed, sonic_ports_default_mtu and sonic_ports_default_fec
+
+All these variables are now gathered in the dictionary `sonic_config_ports`.
+
+Change
+
+```yaml
+sonic_ports_default_speed: 100000
+sonic_ports_default_mtu: 9000
+sonic_ports_default_fec: none
+
+sonic_ports:
+  - fec: rs
+    ips:
+      - 10.0.0.1/32
+    mtu: 1420
+    name: Ethernet0
+    speed: 40000
+    vrf: Vrf45
+```
+
+to
+
+```yaml
+sonic_config_ports:
+  # New field to configure default value for auto negotiation.
+  default_autoneg: # on
```

Comment:
Why on is commented out?

Reply:
Auto negotiation config is a new feature of this role which doesn't exist in the `sonic` role. The migration guide shows what changes need to be made to adapt to the new role. Since auto negotiation wasn't configured before it is documented here but with a comment to avoid the misunderstanding that `default_autoneg: on` is required when migrating to the new role.

Reply:
EDIT: removed all comments